src/quick-lint-js/parse.h

```diff
@@ -2463,6 +2471,11 @@ class parser {
   template <QLJS_PARSE_VISITOR Visitor>
   void parse_and_visit_if(Visitor &v) {
     QLJS_ASSERT(this->peek().type == token_type::kw_if);
+    depth_guard guard = this->enter_depth();
+    this->depth_++;
+    if (this->depth_ > this->stack_limit) {
+      QLJS_PARSER_UNIMPLEMENTED();
+    }
```

Comment:
Blocking: You're incrementing `this->depth_` twice. This is wrong. Delete this code and rely on `depth_guard`:
```suggestion
    depth_guard guard = this->enter_depth();
```

My assertion suggestion on line 3425 would detect this (I think).